src/coreclr/jit/copyprop.cpp

```diff
@@ -351,19 +395,11 @@ void Compiler::optBlockCopyProp(BasicBlock* block, LclNumToGenTreePtrStack* curS
                     continue;
                 }
 
-                // If we encounter first use of a param or this pointer add it as a live definition.
-                // Since they are always live, we'll do it only once.
-                if (lvaGetDesc(lclNum)->lvIsParam || (lclNum == info.compThisArg))
+                // If we encounter first use of a param or this pointer add it as a
+                // live definition. Since they are always live, we'll do it only once.
+                if ((lvaGetDesc(lclNum)->lvIsParam || (lclNum == info.compThisArg)) && !curSsaName->Lookup(lclNum))
```

Comment:
I might have to change this for "hidden buffer" work. I will have to double check once you merge this PR.

Reply:
I believe you will not need to change this particular bit, it is sort of a creative way to avoid pushing unused parameters on the def stack (actual SSA renaming pushes them always).

You will need to change the path that looks for actual definitions though (`ASG`s). I think the best way to do that would be to modify `DefinedLocal` to return `true` for the relevant `CALL`s (everything else will then work as-is).